### ISSUE TITLE
Add a function to get all Control nodes at a specific position

### ISSUE THREAD

**Issue by popcar2** [godotengine/godot-proposals#8408]
### Describe the project you are working on

I'm creating an OS-like interface in Godot. I want to show a context menu when the player right clicks the mouse. Depending on what the player right clicks, it would show different options in the context menu.

The context menu is an autoload and I'd prefer if all the right click handling code was inside of it. 

### Describe the problem or limitation you are having in your project

There is no way to get the Control node(s) that the mouse right clicked. A Control node can only know when it got right clicked itself, but there's no way to globally know which Control nodes in general were clicked.

For example if I want a folder, taskbar, and window to show different context menus when right clicked, I would have to write code in each one of them to handle the right click then signal it to my context menu autoload. This isn't very intuitive or scalable. 

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

A global function to get the Control nodes under the mouse while respecting `mouse_filter` rules.

It could also be a more generic function to get Control nodes at any position (where I can then give it `get_global_mouse_position()`).

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

```swift
func _input(event):
    if event is InputEventMouseButton and event.button_index == 2:
        var mouse_pos = get_global_mouse_position()
        var control_nodes = Input.get_control_nodes_at_pos(mouse_pos)
        for node in control_nodes:
            # Do something here
```

### If this enhancement will not be used often, can it be worked around with a few lines of script?

It's hackier than a usual workaround, but I [added a component that handles right clicking of each node in a specific group](https://godotforums.org/d/37467-is-there-a-global-way-to-get-the-control-node-im-clicking/11). This isn't intuitive or performant.

### Is there a reason why this should be core and not an add-on in the asset library?

Knowing what you clicked and haven't clicked is commonly needed when developing advanced UI. I commonly see people ask questions like "How can I tell which Control node I clicked?" and "How can I know when the user has clicked OUTSIDE of a button?", there isn't a simple way of doing this without workarounds.

A popular workaround is creating a `Rect2` the same size and position as a Control node and seeing if the mouse event is inside it, but this doesn't respect `mouse_filter` so you'd have to implement code checking if the mouse is over it anyways.

In conclusion, I think being able to know which Control nodes got clicked and which ones weren't is very useful when building complex UI. Being able to get Control nodes at a given position is a good, generic way of creating this mechanic.

**Comment by Spartan322** [comment]
Why not expose [Viewport::gui_find_control](https://github.com/godotengine/godot/blob/13a0d6e9b253654f5cc2a44f3d0b3cae10440443/scene/main/viewport.cpp#L1654)?

**Comment by Zireael07** [comment]
I think it's what was exposed in #85966  ?

**Comment by AThousandShips** [comment]
No it's what's being exposed in:
* https://github.com/godotengine/godot/pull/86343

**Comment by Spartan322** [comment]
There is one case about this I can in which you have multiple overlapping controls, this only gets somewhat solved by that to be fair.

**Comment by pc9098** [comment]
Right now I have multiple overlapping controls, this solution would be best for me

**Comment by bfloch** [comment]
The other use-case is debugging overlapping controls that steal mouse inputs.

This proposal or  https://github.com/godotengine/godot/pull/103929 would help to quickly track down which Controls are in the way. Thanks for bringing it up.
